### PR TITLE
Add smoke tests

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-842

### Change description ###
Add smoke tests:
- pay hub health
- test if messages can be read from payments queue

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/smokeTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/PaymentsQueueHealthTest.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import static org.assertj.core.api.Assertions.assertThatCode;
+
+@TestPropertySource("classpath:application.conf")
+class PaymentsQueueHealthTest {
+
+    private static final String CONNECTION_STRING = ConfigFactory.load()
+        .getString("payments-queue-read-connection-string");
+
+    @Test
+    void can_read_messages_from_payments_queue() throws ServiceBusException, InterruptedException {
```
Comment (nitinprabhu) [inline src/smokeTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/PaymentsQueueHealthTest.java]: `can_peek_messages_from_payments_queue` ?

File: src/smokeTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/PayHubApiHealthTest.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+class PayHubApiHealthTest {
+
+    private static final String TEST_PAY_HUB_URL = ConfigFactory.load().getString("test-pay-hub-url");
+
+    @Test
+    void pay_hub_health_check() {
+
+        RestAssured
+            .given()
+            .baseUri(TEST_PAY_HUB_URL)
```
Comment (doncem) [inline src/smokeTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/PayHubApiHealthTest.java]: we used to have relaxed https flag on here. does it not apply here in this scenario?

File: src/smokeTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/PaymentsQueueHealthTest.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+                new ConnectionStringBuilder(CONNECTION_STRING),
+                ReceiveMode.PEEKLOCK
+            );
+
+            assertThatCode(paymentsReceiver::peek).doesNotThrowAnyException();
+        } finally {
+            if (paymentsReceiver != null) {
+                paymentsReceiver.close();
+            }
+        }
```
Comment (doncem) [inline src/smokeTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/PaymentsQueueHealthTest.java]: urgh, microsoft and their non-autocloseable closeable interfaces 🤦‍♂

File: src/smokeTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/PayHubApiHealthTest.java (hunk 2 above)
Comment (aliveni) [reply src/smokeTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/PayHubApiHealthTest.java]: I removed it when trying to test in local. Added it back now. thanks

File: src/smokeTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/PayHubApiHealthTest.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import org.junit.jupiter.api.Test;
+import org.springframework.test.context.TestPropertySource;
+import uk.gov.hmcts.reform.logging.appinsights.SyntheticHeaders;
+
+import static org.hamcrest.Matchers.is;
+
+@TestPropertySource("classpath:application.conf")
+class PayHubApiHealthTest {
+
+    private static final String TEST_PAY_HUB_URL = ConfigFactory.load().getString("test-pay-hub-url");
```
Comment (rkondratowicz) [inline src/smokeTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/PayHubApiHealthTest.java]: we should check health of the service what we're deploying

Comment (nitinprabhu) [reply src/smokeTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/PayHubApiHealthTest.java]: Deploy stage in pipeline will do the health check for app ?

Comment (rkondratowicz) [reply src/smokeTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/PayHubApiHealthTest.java]: remove it then

Comment (rkondratowicz) [reply src/smokeTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/PayHubApiHealthTest.java]: well... if this apps health check verifies payhub

**Comment by aliveni** [comment]
> Check health of bulk-scan-payment-processor and that **bulk-scan-payment-processor** can read from the queue.
> Whether test can reach payhub and queue or not is really relevant.

@rkondratowicz  Updated the pr as you suggested

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/health/PaymentsQueueHealthIndicator.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import com.microsoft.azure.servicebus.IMessageReceiver;
+import com.microsoft.azure.servicebus.primitives.ServiceBusException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.stereotype.Component;
+
+@Component
+public class PaymentsQueueHealthIndicator implements HealthIndicator {
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/health/PaymentsQueueHealthIndicator.java]: Ideally detail of health could be called just `queue`. So I think we should rename this class to `QueueHealthIndicator` - doesn't really matter what kind of queue it checks for the health endpoint perspective

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/health/PaymentsQueueHealthIndicator.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        this.messageReceiver = messageReceiver;
+    }
+
+    @Override
+    public Health health() {
+        try {
+            messageReceiver.peek();
+            messageReceiver.close();
+            return Health.up().build();
+        } catch (InterruptedException | ServiceBusException e) {
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/health/PaymentsQueueHealthIndicator.java]: `InterruptedException` should add extra step getting current thread and explicitly calling interrupt on it

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/health/PaymentsQueueHealthIndicator.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+            log.error("Error occurred while reading messages from payments queue", e);
+            return Health.down().build();
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/health/PaymentsQueueHealthIndicator.java]: do we want to build with `Health.down(e)` instead?

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/health/PayHubHealthIndicator.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    private Health statusDown() {
+        return Health.down().build();
+    }
+
+    private Health statusUnknown() {
+        return Health.unknown().build();
+    }
+}
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/health/PayHubHealthIndicator.java]: Ah, same in this class - `down(exception)`?

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/config/RestTemplateConfiguration.java [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+        return new ApacheHttpClient(getHttpClient());
+    }
+
+    @Bean
+    public RestTemplate restTemplate() {
+        return new RestTemplate(clientHttpRequestFactory());
+    }
+
+    @Bean
+    public HttpComponentsClientHttpRequestFactory clientHttpRequestFactory() {
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/config/RestTemplateConfiguration.java]: should this be a bean?

**Pull request merged**